src/useListState/index.js

```
@@ -10,10 +15,10 @@ export const useListState = ({
 } = {}) => {
   const checkItemHasKey = item =>
     !item.key ? throwError('List item must have a key value') : item;
-  const [listState, setState] = useStateDep(initialValue.map(checkItemHasKey));
+  const [objectListState, setObjectState] = useStateDep(initialValue.map(checkItemHasKey).reduce(reduceArrayToObject, {}));
```

Instead of looping over the array twice I would throw this error in the reduce.